src/management/moderation/moderation-manager.service.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+/**
+ * @ngdoc service
+ * @name udb.management.moderation
+ * @description
+ * # Moderation Manager
+ * This service allows you to lookup moderation lists and approve/reject/... Offers.
+ */
+angular
+  .module('udb.management.moderation')
+  .service('ModerationManager', ModerationManager);
+
+/* @ngInject */
+function ModerationManager(udbApi, OfferWorkflowStatus, jobLogger, BaseJob, $q) {
+  var service = this;
+
+  /**
+   * @return {Promise.<Role[]>}
+   */
+  service.getMyRoles = function() {
+    return udbApi
+      .getMe()
+      .then(function(user) {
+        return udbApi.getUserRoles(user.id);
+      });
+  };
+
+  /**
+   * Find moderation items
+   *
+   * @param {string} queryString
+   * @param {int} itemsPerPage
+   * @param {int} offset
+   *
+   * @return {Promise.<PagedCollection>}
+   */
+  service.find = function(queryString, itemsPerPage, offset) {
+    queryString = (queryString ? queryString + ' AND ' : '') + 'wfstatus="readyforvalidation"';
+
+    return udbApi
+      .findEventsWithLimit(queryString, offset, itemsPerPage);
+  };
+
+  /**
+   * @param {string} offerId
+   *
+   * @return {Promise.<Offer>}
+   */
+  service.getModerationOffer = function(offerId) {
+    return udbApi.getOffer(new URL(offerId));
+  };
+
+  /**
+   * @param {UdbPlace|UdbEvent} offer
+   *
+   * @return {Promise.<BaseJob>}
+   */
+  service.approve = function(offer) {
+    return udbApi
+      .patchOffer(offer['@id'], 'Approve')
+      .then(logRoleJob);
+  };
+
+  /**
+   * @param {UdbPlace|UdbEvent} offer
+   *
+   * @return {Promise.<BaseJob>}
+   */
+  service.reject = function(offer, reason) {
+    return udbApi
+      .patchOffer(offer['@id'], 'Reject', reason)
+      .then(logRoleJob);
+  };
+
+  /**
+   * @param {UdbPlace|UdbEvent} offer
+   *
+   * @return {Promise.<BaseJob>}
+   */
+  service.flagAsDuplicate = function(offer) {
```

Comment:
@nick De naam is aangepast om consistent te zijn met de commands op de backend. Het is zo ook duidelijk dat je een aanbod simpelweg weigert en geen andere actie op doet zoals _duplicate_ of _inappropriate_.